analysis/src/SignatureHelp.ml

```diff
@@ -154,9 +154,68 @@ let signatureHelp ~path ~pos ~currentFile ~debug =
   let posBeforeCursor = Pos.posBeforeCursor pos in
   let supportsMarkdownLinks = true in
   let foundFunctionApplicationExpr = ref None in
-  let setFound r = foundFunctionApplicationExpr := Some r in
+  let setFound r =
+    if !foundFunctionApplicationExpr = None then
+      foundFunctionApplicationExpr := Some r
+  in
+  let searchForArgWithCursor ~isPipeExpr ~args ~exp =
```

Comment:
This whole function body is just moved in here from below, with very minor changes.